src/pvrclient-mythtv.cpp

```diff
@@ -1666,12 +1666,14 @@ PVR_ERROR PVRClientMythTV::AddTimer(const PVR_TIMER &timer)
   XBMC->Log(LOG_DEBUG, "%s: title: %s, start: %ld, end: %ld, chanID: %u", __FUNCTION__, timer.strTitle, timer.startTime, timer.endTime, timer.iClientChannelUid);
   CLockObject lock(m_lock);
   // Check if our timer is a quick recording of live tv
-  // Assumptions: Timer start time = 0, and our live recorder is locked on the same channel.
+  // Assumptions: Our live recorder is locked on the same channel and the recording starts
+  // at the same time as or before (includes 0) the currently in progress program
   // If true then keep recording, setup recorder and let the backend handle the rule.
-  if (timer.startTime == 0 && m_liveStream && m_liveStream->IsPlaying())
+  if (m_liveStream && m_liveStream->IsPlaying())
   {
     Myth::ProgramPtr program = m_liveStream->GetPlayedProgram();
-    if (timer.iClientChannelUid == FindPVRChannelUid(program->channel.chanId))
+    if (timer.iClientChannelUid == FindPVRChannelUid(program->channel.chanId) &&
+        timer.startTime <= program->startTime)
```

Comment:
Actually no it won't the way you suggest. We need a 100% epg match, or the 
user won't get what they expect.
Andrew

Sent with AquaMail for Android
http://www.aqua-mail.com


On June 30, 2016 5:17:47 PM Jean-Luc Barrière <notifications@github.com> wrote:

> >    {
> >      Myth::ProgramPtr program = m_liveStream->GetPlayedProgram();
> > -    if (timer.iClientChannelUid == 
> FindPVRChannelUid(program->channel.chanId))
> > +    if (timer.iClientChannelUid == 
> FindPVRChannelUid(program->channel.chanId) &&
> > +        timer.startTime <= program->startTime)
>
> Is endTime more suitable ?
>
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/janbar/pvr.mythtv/pull/39/files/0c71e0f8e71ace31656d19b9174ec374241dcab1#r69163441